packages/reporting/test/country-provider.spec.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * WhoTracks.Me
+ * https://whotracks.me/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import { expect } from 'chai';
+
+import CountryProvider from '../src/country-provider.js';
+
+const SECOND = 1000;
+const MINUTE = 60 * SECOND;
+const HOUR = 60 * MINUTE;
+const DAY = 24 * HOUR;
+const WEEK = 7 * DAY;
+
+// Some big value which should be large enough to
+// make sure all cooldowns have expired.
+const SKIP_ALL_COOLDOWNS = WEEK;
+
+describe('#CountryProvider', function () {
+  const storageKey = 'some-storage-key';
+  const apiEndpoint = 'https://some-endpoint.test';
+  let uut;
+  let config;
+  let storage;
+  let fetchMock;
+
+  function mockApi() {
+    const mock = async (url) => {
+      expect(url).to.eql(apiEndpoint);
+      mock.numRequests += 1;
+      return fetchMock.response();
+    };
+    mock.response = () => new Error('Not configured');
+    mock.numRequests = 0;
+    return mock;
+  }
+
+  function assumeApiReturns(ctry) {
+    fetchMock.response = async () => ({
+      ok: true,
+      async json() {
+        return {
+          location: ctry,
+          'location.city': '--',
+          ts: '20221108',
+        };
+      },
+    });
+  }
+
+  function assumeApiIsDown() {
+    mockApi(async () => ({
+      ok: false,
+      statusText: 'Stub server has been configured to fail (this is expected).',
+    }));
+  }
+
+  function newCountryProvider() {
+    return new CountryProvider({
+      config,
+      storage,
+      storageKey,
+      _fetchImpl: fetchMock,
```

Comment:
cannot we use something link this https://github.com/whotracksme/webextension-packages/blob/main/packages/anonymous-communication/test/server-public-key-accessor.spec.js#L79 instead?

Reply:
OK, I'll update the PR to remove the dependency injection by mocking.